### PR TITLE
fish: Add fish shell package

### PR DESCRIPTION
**Pull request by InBetweenNames**
Maintainer: Shane Peelar / @<InBetweenNames>
Compile tested: AMD64
Run tested: bcm53xx, RT-AC87U, LEDE snapshots January 30th

Description:

These two commits add the PCRE2 package and the fish package.  Since fish depends on PCRE2, the former commit is necessary to make it work.

### CODE REVIEW

**Comment by diizzyy** [comment]
I'm all for new interesting packages but I don't see the need for this on very "small"/limited platforms. Care to elaborate on potential practical benefits?

**Comment by InBetweenNames** [comment]
A user experience benefit, really.  Fish doesn't consume that many more resources than bash does and it's running just fine on my router currently.  I'm currently hosting my blog using OpenWRT/LEDE (I alternate) and having fish makes my life a lot easier for doing maintenance tasks.  I have no problem maintaining this out of tree if you don't want it in the master branch, I just figured I'd contribute back something I found useful.

**Comment by thess** [comment]
Speaking of shells ... We also have Midnight Commander (aka _mc_). So I see no problem with another.

However, I'd like to see separate PRs for each package even though pcre2 is a necessary dependent - You can point that out in your new fish PR. I will also have some review comments shortly on your Makefiles.

File: utils/fish/Makefile [hunk 1]
```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (C) 2007-2015 OpenWrt.org
```
Comment (thess) [inline utils/fish/Makefile]: Remove copyright or replace with yourself.

File: utils/fish/Makefile [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+PKG_SHA1SUM:=c5981245b45ea61b765008299b5ac477657089c3
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(BASE_VERSION)
+
+PKG_LICENSE:=GPL-3.0+
+PKG_LICENSE_FILES:=COPYING
+PKG_MAINTAINER:=Shane Peelar <lookatyouhacker@gmail.com>
+
+PKG_FIXUP:=autoreconf
+
+PKG_CHECK_FORMAT_SECURITY:=0
```
Comment (thess) [inline utils/fish/Makefile]: Is there a build problem with this on?

File: utils/fish/Makefile [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  fish is a user friendly commandline shell intended mostly for interactive use. 
+	For the latest information on fish, please visit the fish homepage.
+endef
+
+
+define Build/Configure
+	$(call Build/Configure/Default, \
+		--without-included-pcre2)
+endef
+
```
Comment (thess) [inline utils/fish/Makefile]: Replace with simple
>CONFIG_ARGS+=--without-included-pcre2

File: utils/fish/Makefile [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+define Build/Compile
+	$(MAKE) -C $(PKG_BUILD_DIR) \
+		DESTDIR="$(PKG_INSTALL_DIR)" \
+		SHELL="/bin/sh" \
+		all install
+endef
+
```
Comment (thess) [inline utils/fish/Makefile]: Replace with:
TARGET_MAKE_FLAGS+='SHELL="/bin/sh"' and set PKG_INSTALL:=1

**Comment by InBetweenNames** [comment]
Hi thess,

Thanks for your comments, I have updated the Makefile to include your suggestions.  If everything looks good, I can separate this into two different PRs: one for PCRE2 and one for fish.

**Comment by thess** [comment]
You really need to make these 2 packages into separate PRs before we go any further. Additionally...

See my updated comments about CONFIGURE_ARGS, and MAKE_FLAGS etc.
PKG_INSTALL should be defined before including _package.mk_

fish - I Couldn't build it without adding: `EXTRA_CXXFLAGS+= -std=gnu++11`
pcre2 - update to latest 10.22 and include PKG_HASH with SHA256SUM as well as correct PKG_MD5HASH.

Also, you need squash/rebase each of your updates to a single commit.

**Comment by InBetweenNames** [comment]
Hi thess, thank you very much for your feedback.  I've incorporated your changes and have split the commits accordingly across two PRs:

pcre2: https://github.com/openwrt/packages/pull/4002
fish: https://github.com/openwrt/packages/pull/4003

With these changes, I'm closing this PR and I look forward to continuing discussions on the other two PRs.

**Comment by UniverseXXX** [comment]
Was fish finally moved to the master branch and released? I couldn't find it in any repository. 
Thanks.

**Pull request closed without merging**